lib/rubocop/cop/style/random_with_offset.rb

```diff
@@ -0,0 +1,179 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for the use of randomly generated numbers,
+      # added/subtracted with integer literals, as well as those with #succ
+      # and #pred methods. Prefer using ranges instead, as it clearly
+      # states the intentions.
+      #
+      # @example
+      #   # bad
+      #   rand(6) + 1
+      #
+      #   # bad
+      #   1 + rand(6)
+      #
+      #   # bad
+      #   rand(6) - 1
+      #
+      #   # bad
+      #   1 - rand(6)
+      #
+      #   # bad
+      #   rand(6).succ
+      #
+      #   # bad
+      #   rand(6).pred
+      #
+      #   # bad
+      #   Random.rand(6) + 1
+      #
+      #   # bad
+      #   Kernel.rand(6) + 1
+      #
+      #   # bad
+      #   rand(0..5) + 1
+      #
+      #   # good
+      #   rand(1..6)
+      #
+      #   # good
+      #   rand(1...7)
+      class RandomWithOffset < Cop
+        MSG = 'Prefer ranges when generating random numbers instead of ' \
+          'integers with offsets.'.freeze
+
+        def_node_matcher :integer_op_rand?, <<-PATTERN
+          (send
+            int {:+ :-}
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange}))
+        PATTERN
+
+        def_node_matcher :rand_op_integer?, <<-PATTERN
+          (send
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange})
+            {:+ :-}
+            int)
+        PATTERN
+
+        def_node_matcher :rand_modified?, <<-PATTERN
+          (send
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange})
+            {:succ :pred :next})
+        PATTERN
+
+        def on_send(node)
+          unless integer_op_rand?(node) || rand_op_integer?(node) ||
```

Comment:
unless -> if and put the add_offense in it. The code looks pretty odd right now. We should have a cop for this. :-)

Reply:
I think this would normally be picked up by `Style/GuardClause`, but because:

```ruby
return unless integer_op_rand?(node) || rand_op_integer?(node) || rand_modified?(node)
```

exceeds the maximum line length, it is silenced.

For OP there are some more options, one is to break the condition:

```ruby
return unless integer_op_rand?(node) ||
              rand_op_integer?(node) ||
              rand_modified?(node)
```

Another is to combine the node patterns:

```ruby
def_node_matcher :offending_rand?, <<-PATTERN
  {integer_op_rand? rand_op_integer? rand_modified?}
PATTERN
```

Reply:
The second solution for some reason doesn't work with me:

```
      Failure/Error: {integer_op_rand? rand_op_integer? rand_modified?}

      NoMethodError:
        undefined method `integer_op_rand?' for #<RuboCop::AST::SendNode:0x005558416338c0>
```

Reply:
@donjar: I think I forgot a leading `#` on all the funcalls. 😅